previews/model.py

```
@@ -56,71 +58,102 @@ def fetch(self, timeout=1):
-                break
-            else:
-                # Get the <title> as a string
-                self.title = str(soup.title.string)
-
-        titles = re.compile("[-–|:•]+").split(self.title)
-
-        self.title = titles[0].strip()
-
-        # Get the desc from whatever we can find
-        desc_elems = soup.findAll("meta", attrs={"name": re.compile(r"Desc", re.I)})
-
-        for meta_elem in desc_elems:
-            if meta_elem.attrs["content"]:
-                self.desc = meta_elem.attrs["content"]
-                break
-
-        if len(self.desc.split()) > 30:
-            self.desc = " ".join(self.desc.split()[0:29]).strip()
-
-            self.desc = self.desc.strip("…")
-            self.desc = self.desc.strip(".")
-            self.desc += "..."
-
-        icon_link = soup.find("link", rel=re.compile(r"shortcut icon"))
-
-        if icon_link is None:
-            icon_link = soup.find("link", rel=re.compile(r"icon"))
-
-        if icon_link is not None:
-            # Check if icon link is global or relative
-            icon_href = icon_link["href"]
-
-            if icon_href.find("http") != -1:
-                self.icon = icon_href
-            else:
-                self.icon = self.url + icon_href
-
-        # Fetch Open Graph Image
-        image = soup.find("meta", property="og:image")
-
-        if image is None:
-            # Use favicon if no image is specified
-            self.image = self.icon
-
-        if image is not None:
-            # Check if image link is global or relative
-            image_link = image["content"]
-
-            if image_link.find("http") != -1:
-                self.image = image_link
-            else:
-                self.image = self.url + image_link
-
-            self.image = builder.create_url(
-                self.image,
-                {'max-w': IMAGE_LINK_MAX_WIDTH}
-            )
+        if soup is None:
+            raise Exception("Unable to fetch content")
```

Commented this earlier but I think it fell through the cracks. Can we please raise a custom exception instead? It's just easier for debugging while looking at this in sentry or another tool
e.g. we can see how often this happens with a custom exception type but not with the generic one.

---

Ah gotcha - ok done I've made a custom exception for this called `ValueIsNoneError`